service/frontend/workflow_handler.go

```diff
@@ -3730,16 +3728,160 @@ func (wh *WorkflowHandler) ListBatchOperations(
 	}, nil
 }
 
-func (*WorkflowHandler) PollNexusTaskQueue(context.Context, *workflowservice.PollNexusTaskQueueRequest) (*workflowservice.PollNexusTaskQueueResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "unimplemented")
+func (wh *WorkflowHandler) PollNexusTaskQueue(ctx context.Context, request *workflowservice.PollNexusTaskQueueRequest) (_ *workflowservice.PollNexusTaskQueueResponse, retError error) {
```

Comment:
I'd appreciate unit tests for the new functionality you're adding, not just integration tests

Reply:
I don't think unit tests are practical for a gRPC handler.

I will add more functional test cases to make sure I cover all of the conditions in this function.
I will add some unit tests for the matching engine code if I feel that I can't cover everything with functional tests.

Reply:
We already have them for many of these handlers in workflow_handler_test.go. I mostly appreciate them so we can very quickly test failure paths without needing to wait on the DB, etc..

I won't hold the PR up because of that though; I'll take some tests over no tests!